### PR TITLE
Perform interaction success came instead of unsupported resource

### PR DESCRIPTION
**Pull request by OHerasym**
Add UNSUPPORTED_RESOURCE resultCode in PerformInteraction

Related:
             [APPLINK-17490](https://adc.luxoft.com/jira/browse/APPLINK-17490)

### CODE REVIEW

**Comment by OHerasym** [comment]
Please review: @dtrunov , @malirod , @AMikhniev , @VVeremjova, @Kozoriz

File: src/components/application_manager/src/commands/mobile/perform_interaction_request.cc [hunk 1]
```diff
@@ -425,17 +424,21 @@ void PerformInteractionRequest::ProcessPerformInteractionResponse(
...
-  const char* return_info = (info.empty()) ? NULL : info.c_str();
   const SmartObject* response_params =
       msg_params.empty()
       ? NULL
       : &msg_params;
-  SendResponse(result, result_code, return_info, response_params);
+
+  if (mobile_apis::InteractionMode::BOTH != interaction_mode_) {
+    DisablePerformInteraction();
+    SendResponse(ui_result_, ui_resultCode_, info.c_str(), response_params);
```
Comment (dtrunov) [inline src/components/application_manager/src/commands/mobile/perform_interaction_request.cc]: Please clarify, we need waiting for response from VR and UI if SDL receives interaction mode VR and MANUAL_ONLY.

File: src/components/application_manager/src/commands/mobile/perform_interaction_request.cc [hunk 2]
```diff
@@ -220,13 +227,19 @@ void PerformInteractionRequest::on_event(const event_engine::Event& event) {
     case hmi_apis::FunctionID::UI_PerformInteraction: {
       LOG4CXX_DEBUG(logger_, "Received UI_PerformInteraction event");
       ui_response_recived_ = true;
-      ProcessPerformInteractionResponse(event.smart_object());
+      unsubscribe_from_event(hmi_apis::FunctionID::UI_PerformInteraction);
+      ui_resultCode_ = GetMobileResultCode(static_cast<hmi_apis::Common_Result::eType>(
+          message[strings::params][hmi_response::code].asUInt()));
+      ProcessPerformInteractionResponse(event.smart_object(), msg_param);
```
Comment (VVeremjova) [inline src/components/application_manager/src/commands/mobile/perform_interaction_request.cc]: Why did not you use `message` from line 216?

**Comment by AMikhniev** [comment]
Reviewed

**Comment by malirod** [comment]
Rebase, please

**Comment by OHerasym** [comment]
Please review last commit: @malirod

**Comment by malirod** [comment]
@OHerasym, fix the title of the PR. Remove hotfix.

**Comment by OHerasym** [comment]
@malirod , Removed hotfix

**Comment by malirod** [comment]
@anosach-luxoft, @AGaliuzov please review. There are changes in the business logic

**Comment by malirod** [comment]
Reviewed

**Comment by Kozoriz** [comment]
Reviewed

**Comment by AMikhniev** [comment]
Reviewed

File: src/components/application_manager/src/commands/mobile/create_interaction_choice_set_request.cc [hunk 3]
```diff
@@ -117,75 +117,32 @@ void CreateInteractionChoiceSetRequest::Run() {
 
 mobile_apis::Result::eType CreateInteractionChoiceSetRequest::CheckChoiceSet(
   ApplicationConstSharedPtr app) {
+  using namespace smart_objects;
   LOG4CXX_AUTO_TRACE(logger_);
 
+  std::set<uint32_t> choice_id_set;
+
   const smart_objects::SmartArray* new_choice_set_array =
```
Comment (VVeremjova) [inline src/components/application_manager/src/commands/mobile/create_interaction_choice_set_request.cc]: If you added namespace 'smart_objects', why did you leave `smart_objects` here?

**Comment by VVeremjova** [comment]
Reviewed

**Pull request merged**